File: 61. Rotate List/Solution.java
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        
        if (head == null || head.next == null || k == 0) return head;
        
        int len = 0;
        
        ListNode pHead = head;
        ListNode pLast = null;
        
        while (pHead != null) {
            len++;
            pLast = pHead;
            pHead = pHead.next;
        }
        
        int kMod = k % len;
        
        if (kMod == 0) return head;
        
        pHead = head;
        for (int i=1; i<len-kMod; i++) {
            pHead = pHead.next;
        }
        ListNode newHead = pHead.next;
        pLast.next = head;
        pHead.next = null;
        
        return newHead;
    }
}

File: 61. Rotate List/Solution.py
class Solution:
    def rotateRight(self, head: ListNode, k: int) -> ListNode:
        
        if head is None or head.next is None or k == 0:
            return head
        
        listLength = 0;
        
        pHead = head
        pLast = None
        
        while pHead is not None:
            listLength += 1
            pLast = pHead
            pHead = pHead.next
            
        kMod = k % listLength
        
        if kMod == 0:
            return head
        
        pLast.next = head
        pHead = head
        
        for i in range(1,listLength-kMod):
            pHead = pHead.next
            
        newHead = pHead.next
        pHead.next = None
        
        return newHead
        